### PR TITLE
feat: USB hotswapping for replay storage

### PR DESCRIPTION
**Pull request by jmlee337**
the general methodology of this change is as follows

1. copy the technique from HID.c to set up a thread and addition to main loop to manage the IOCTLs and keep track of devices being inserted or removed
2. copy enough of libogc/usb.c and libogc/usbstorage.c in to mount inserted devices
3. modify SlippiFileWriter logic to enable hotswap for usb

caveat: as currently implemented, the game will still not boot without the usb replay device inserted. I am working on removing this limitation as a follow-up.

another note: this may require some closed testing before wide use. I encountered some game crashes while implementing. However I believe those were attributable solely to errors at the time. Additionally the crashes always occurred as the game started (between stage select and go) and not during the middle of games

please feel free to @ me in discord if there are any questions or explanations required

### CODE REVIEW

**Comment by jmlee337** [comment]
Here's a sample of aforementioned game crash btw:
```
HSD ArchiveParse: byte-order mismatch! Please check data format 20004002 56
HSD ArchiveParse error!
assertion "0" failed in lbarchive.c on line 73.
DATE Feb 13 2002 TIME 22:06:27

SN v1.11.1-5-9bc7eda8-dirty
 Console runtime: 2415 frames
 Scene runtime: 144 frames

- STACK -----------------------
 Address:  Back Chain  LR Save
804EE9A8:   804EE9D0   803882EC
804EE9D0:   804EE9F0   80388260
804EE9F0:   804EEA18   80016AA4
804EEA18:   804EEAB8   80016D20
804EEAB8:   804EEB10   800222E4
804EEB10:   804EEB30   8016E7C8
804EEB30:   804EEB48   8016E94C
804EEB48:   804EEB80   801A40E8
804EEB80:   804EEBB0   801A44C4
804EEBB0:   804EEBD0   801A45C8
804EEBD0:   804EEBF8   801601AC
804EEBF8:   FFFFFFFF   8000533C
```

PXL_20230926_140715561

**Comment by jmlee337** [comment]
squashed commits.

To reiterate from discord, I've had testers running weeklies for 10 months and larger locals (up to 64 entrants) for 3 months. No regressions or confirmed bugs have been found

File: loader/source/global.c [hunk 1]
```diff
@@ -476,6 +451,24 @@ void UpdateNinCFG()
...
+				ncfg->MeleeCodeOptions[0] = 2; // UCF
+				break;
+			default:
+				ncfg->MeleeCodeOptions[0] = 1; // Off
+		}
+		ncfg->Version = 0xC;
+	}
+	if (ncfg->Version == 0xC)
+	{
+		// 0 defaults are fine
```
Comment (NikhilNarayana) [inline loader/source/global.c:469]: check if replays are currently enabled and set the LED to stealth if so

File: kernel/SlippiFileWriter.c [hunk 2]
```diff
@@ -177,16 +207,53 @@ static u32 SlippiHandlerThread(void *arg)
 
 	u32 writtenByteCount = 0;
 	driveTimer = read32(HW_TIMER);
+	driveTimerSet = false;
+
+	FATFS device;
+	bool failedToMount = false;
+	bool hasFile = false;
+	bool mounted = true;
+	const bool use_usb = ConfigGetUseUSB() != 1;
```
Comment (NikhilNarayana) [inline kernel/SlippiFileWriter.c:214]: `u32 UseUSB = ConfigGetUseUSB(); // Returns 0 for SD, 1 for USB`

based on the line above, shouldn't this be `const bool use_usb = ConfigGetUseUSB() (== 1);`?

**Review by NikhilNarayana — COMMENTED** [review]
some quick thoughts while i get my head around the rest

File: kernel/SlippiFileWriter.c (hunk 2 above)
Comment (jmlee337) [reply kernel/SlippiFileWriter.c:214]: `ConfigGetUseUSB` refers to ISO device, so file write device is 'opposite'

**Comment by jmlee337** [comment]
Removed "stealth" Replays LED setting as discussed in discord
As well see below, Replays LED setting description with the existing UCF setting description for comparison
PXL_20240925_120155904
PXL_20240925_085314432

File: loader/source/menu.c [hunk 3]
```diff
@@ -168,13 +168,23 @@ static const char *desc_networking[] = {
...
 	NULL
 };
+static const char *desc_slippi_replays_led[] = {
+	"How Drive LED should be used to",
+	"indicate replay device status",
+	"",
+	"   [Yes]",
+	"    Drive LED will stay lit as long",
+	"    as a replay device is inserted",
+	"    and working properly.",
```
Comment (NikhilNarayana) [inline loader/source/menu.c]:
```suggestion
	"Use the Wii Disc Drive LED",
	"to indicate replay device status.",
	"The LED will remain on while",
	"the replay device is inserted",
	"and working properly.",
```

**Review by NikhilNarayana — APPROVED** [review]

**Pull request merged**